library/core/src/char/methods.rs

```diff
@@ -1090,7 +1093,7 @@ impl char {
     #[stable(feature = "ascii_methods_on_intrinsics", since = "1.23.0")]
     #[inline]
     pub fn to_ascii_uppercase(&self) -> char {
-        if self.is_ascii() { (*self as u8).to_ascii_uppercase() as char } else { *self }
+        if self.is_ascii_lowercase() { ((*self as u8) & !ASCII_CASE_MASK) as char } else { *self }
```

Comment:
u8::to_ascii_lowercase already has a different way of writing the same bit flip, it would of course be nice to re-use it or at least express it in the same way? (Or is that a pointless exercise?). Of course the condition in to_ascii_lowercase means that we can't reuse it straight away.

Reply:
let me see if I can reform the other use - it may be there's no impact on doing so...

Reply:
Should we do a perf run after this change?

Reply:
It looks like we can formulate it as this without changing the instructions: https://godbolt.org/z/9cEfr9 - that way they could all refer to the same const.

Reply:
Ok have commited a change to unify the approach. (I've rerun the benchmark and there is still the same improvement.)

Personally I find it clearer than shifting one 5 times.
Is the const in the right module? I didn't seeee the need to make it public.